image-processing/main.py

```diff
@@ -20,10 +23,43 @@ def trigger_event(event, context):
         None; the output is written to Stackdriver Logging
     """
 
-    print("Event ID: {}".format(context.event_id))
-    print("Event type: {}".format(context.event_type))
-    print("Bucket: {}".format(event["bucket"]))
-    print("File: {}".format(event["name"]))
-    print("Metageneration: {}".format(event["metageneration"]))
-    print("Created: {}".format(event["timeCreated"]))
-    print("Updated: {}".format(event["updated"]))
+    bucket_name = event["bucket"]
+    source_img = event["name"]
+    tmp_download_path = f"/tmp/{source_img}"
+
+    try:
+        download_image(bucket_name, source_img, tmp_download_path)
+        print(f"Image {source_img} downloaded to {tmp_download_path}")
+        # TODO(abi) call cloud vision API with this image
+        # TODO(jerry) call R script with features returned from Abi's part
+        # TODO(hantao) put processed images to `sie-processed-images` bucket
+    except Exception:
+        pass
+    finally:
+        if os.path.isfile(tmp_download_path):
+            os.remove(tmp_download_path)
+        else:
+            print("Error: %s file not found" % tmp_download_path)
+
+
+def download_image(bucket_name, source_blob_name, destination_file_name):
+    """Downloads the specified image from cloud storage
+    Args:
+        bucket_name = "your-bucket-name"
+        source_blob_name = "image-name"
+        destination_file_name = "local/path/to/file"
+
+    Returns:
+        Local path to downloaded image
+    """
+
+    storage_client = storage.Client()
+
+    bucket = storage_client.bucket(bucket_name)
+
+    # Construct a client side representation of a blob.
+    # Note `Bucket.blob` differs from `Bucket.get_blob` as it doesn't retrieve
+    # any content from Google Cloud Storage. As we don't need additional data,
+    # using `Bucket.blob` is preferred here.
+    blob = bucket.blob(source_blob_name)
+    blob.download_to_filename(destination_file_name)
```

Comment:
Does this download_to_filename handle multiple files? I am writing upload image, but I guess I need to handle multiple images upload instead of single, so I am a little confused here.

Reply:
we only need to download one image at a time only. Which part are you confused about?

Reply:
Ok, I think that clarifies my confusion. I will try to make a single image upload.

Reply:
wait just to clarify, your method will put ~400ish images generated from Jerry's function to cloud storage bucket `sie-processed-images`. I'm not sure if there is a bulk-upload option that is faster. If not then yea you probably want to loop uploading single images.

Reply:
Yeah, that is what I was thinking.